### PR TITLE
[create-block] Split documentation into two sections

### PR DESCRIPTION
**Pull request by ryanwelcher**
## What?
The intention of the PR is to separate the documentation for create-block into two sections. The first section is on usage and is the default documentation seen when landing on the page. The second is meant to be a child section and focuses on External Templating which is a much different use case. By splitting these out, we increase readability and allow to user to choose what content they need.

In order to facilitate this, I needed to modify the` docgen` tool to allow for packages to contain their own `toc.json` file that is then ingested when generating the main `manifest.json` file.

### CODE REVIEW

File: docs/tool/manifest.js [hunk 1]
```diff
@@ -31,15 +31,27 @@ const packagePaths = glob( 'packages/*/package.json' )
...
 			markdown_source: path,
 			parent: 'packages',
-		};
-	} );
+		} );
+
+		// Next add any items in the docs/toc.json if found.
+		if ( fs.existsSync( join( __dirname, '..', tocPath ) ) ) {
+			const toc = require( join( __dirname, '..', tocPath ) ).values();
+			for ( const item of toc ) {
```
Comment (gziolo) [inline docs/tool/manifest.js]: Would it work as:

```js
manifest.push( ...toc );
```

I checked https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/push and I see it accepts also more than one element as arguments.

File: packages/create-block/README.md [hunk 2]
```diff
@@ -24,6 +24,8 @@ _(requires `node` version `14.0.0` or above, and `npm` version `6.14.4` or above
 
 It creates a WordPress plugin that you need to [install manually](https://wordpress.org/support/article/managing-plugins/#manual-plugin-installation).
 
+[Watch a video introduction to create-block on Learn.wordpress.org](https://learn.wordpress.org/tutorial/using-the-create-block-tool/)
```
Comment (gziolo) [inline packages/create-block/README.md:27]: Great idea 👍🏻

File: packages/create-block/README.md [hunk 3]
```diff
@@ -144,112 +145,4 @@ _Note: You don’t need to install or configure tools like [webpack](https://web
...
--   `description` (no default) – a short description for your block.
--   `dashicon` (no default) – an icon property thats makes it easier to identify a block ([available values](https://developer.wordpress.org/resource/dashicons/)).
--   `category` (default: `'widgets'`) – blocks are grouped into categories to help users browse and discover them. The categories provided by core are `text`, `media`, `design`, `widgets`, `theme`, and `embed`.
--   `attributes` (no default) – block attributes ([more details](https://developer.wordpress.org/block-editor/developers/block-api/block-attributes/)).
--   `supports` (no default) – optional block extended support features ([more details](https://developer.wordpress.org/block-editor/developers/block-api/block-supports/).
--   `editorScript` (default: `'file:./index.js'`) – an editor script definition.
--   `editorStyle` (default: `'file:./index.css'`) – an editor style definition.
--   `style` (default: `'file:./style-index.css'`) – a frontend and editor style definition.
-
-## Contributing to this package
```
Comment (gziolo) [inline packages/create-block/README.md:249]: I believe we still need this section and everything after it 😄

File: packages/create-block/docs/external-template.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+-   `editorStyle` (default: `'file:./index.css'`) – an editor style definition.
+-   `style` (default: `'file:./style-index.css'`) – a frontend and editor style definition.
+
+## Contributing to this package
+
+This is an individual package that's part of the Gutenberg project. The project is organized as a monorepo. It's made up of multiple self-contained software packages, each with a specific purpose. The packages in this monorepo are published to [npm](https://www.npmjs.com/) and used by [WordPress](https://make.wordpress.org/core/) as well as other software projects.
+
+To find out more about contributing to this package or Gutenberg as a whole, please read the project's main [contributor guide](https://github.com/WordPress/gutenberg/tree/HEAD/CONTRIBUTING.md).
+
+<br /><br /><p align="center"><img src="https://s.w.org/style/images/codeispoetry.png?1" alt="Code is Poetry." /></p>
```
Comment (gziolo) [inline packages/create-block/docs/external-template.md]: That part is necessary only in the README file.

Comment (ryanwelcher) [reply packages/create-block/docs/external-template.md]: Updated

File: packages/create-block/README.md (hunk 3 above)
Comment (ryanwelcher) [reply packages/create-block/README.md:249]: Updated

File: docs/tool/manifest.js (hunk 1 above)
Comment (ryanwelcher) [reply docs/tool/manifest.js]: Updated.

**Comment by gziolo** [comment]
Cool, this is looking great. I will take another look tomorrow 👍🏻

**Review by juanmaguitar — APPROVED** [review]

File: docs/manifest.json [hunk 5]
```diff
@@ -1469,6 +1469,12 @@
 		"markdown_source": "../packages/create-block/README.md",
 		"parent": "packages"
 	},
+	{
+		"title": "@wordpress/create-block External Template",
+		"slug": "packages-create-block-external-template",
+		"markdown_source": "../packages/create-block/README.md",
```
Comment (gziolo) [inline docs/manifest.json]: The source of the file is incorrect.

File: packages/create-block/docs/toc.json [hunk 6]
```diff
@@ -0,0 +1,8 @@
+[
+	{
+		"title": "@wordpress/create-block External Template",
+		"slug": "packages-create-block-external-template",
+		"markdown_source": "../packages/create-block/README.md",
```
Comment (gziolo) [inline packages/create-block/docs/toc.json]:
```suggestion
		"markdown_source": "../packages/create-block/docs/external-template.md",
```

File: docs/manifest.json (hunk 5 above)
Comment (gziolo) [inline docs/manifest.json]:
```suggestion
		"markdown_source": "../packages/create-block/docs/external-template.md",
```

**Review by gziolo — APPROVED** [review]
Excellent addition ❤️ 

It would be great to document how to add more documentation pages for packages based on this PR. There is more packages that would benefit from similar restructuring. The document that covers contributions to the documentation has even a section about adding new documents:

https://github.com/WordPress/gutenberg/tree/trunk/docs/contributors/documentation#create-a-new-document

**Pull request merged**